h/views/api/groups.py

```diff
@@ -95,6 +95,61 @@ def update(group, request):
     return GroupJSONPresenter(GroupContext(group, request)).asdict(expand=['organization'])
 
 
+@api_config(route_name='api.group_upsert',
+            request_method='PUT',
+            permission='upsert',
+            description='Create or update a group')
+def upsert(context, request):
+    """
+    Create or update a group from a PUT payload.
+
+    If no group model is present in the passed ``context`` (on ``context.group``),
+    treat this as a create action and delegate to ``create``.
+
+    Otherwise, replace the existing group's resource properties entirely and update
+    the object.
+
+    :arg context:
+    :type context: h.traversal.contexts.GroupUpsertContext
```

Comment:
This should be `h.traversal.GroupUpsertContext` (the fact that `h.traversal` is broken down into `roots` and `contexts` modules is an internal implementation detail of that package)